### PR TITLE
Improve file name validation

### PR DESCRIPTION
**Pull request by Isaac-DeFrain**
- add property-based testing for block file names

Fixes https://github.com/Granola-Team/mina-indexer/issues/323

### CODE REVIEW

File: src/block/mod.rs [hunk 1]
```diff
@@ -182,24 +182,19 @@ pub fn parse_file(filename: &Path) -> anyhow::Result<PrecomputedBlock> {
...
 pub fn get_state_hash(file_name: &OsStr) -> Option<String> {
     let last_part = file_name.to_str()?.split('-').last()?.to_string();
-    if last_part.starts_with('.') || !last_part.starts_with("3N") {
-        return None;
-    }
-
     let state_hash = last_part.split('.').next()?;
-    if state_hash.contains('.') {
-        return None;
+    if state_hash.starts_with("3N") {
```
Comment (n1tranquilla) [inline src/block/mod.rs:194]: Significance of "3N" ?

File: src/block/mod.rs [hunk 2]
```diff
@@ -214,13 +209,47 @@ pub fn get_blockchain_length(file_name: &OsStr) -> Option<u32> {
...
+    if let Some(ext) = path.extension() {
+        // check json extension
+        if ext.to_str() == Some("json") {
+            // check file stem
+            if let Some(file_name) = path.file_stem() {
+                if let Some(parts) = file_name
+                    .to_str()
+                    .map(|name| name.split('-').collect::<Vec<&str>>())
+                {
+                    if parts.len() == 2 {
```
Comment (n1tranquilla) [inline src/block/mod.rs:224]: is there anyway use `.last()` for `Vec`? Might simplify this. I'm not blocking the MR on this.

File: src/block/mod.rs (hunk 1 above)
Comment (Isaac-DeFrain) [reply src/block/mod.rs:194]: Just some bytes to distinguish state hashes from the other various types of hashes used

Comment (n1tranquilla) [reply src/block/mod.rs:194]: Might it be helpful to capture the type of hash in a variable?

File: src/block/mod.rs (hunk 2 above)
Comment (Isaac-DeFrain) [reply src/block/mod.rs:228]: Great suggestion, much simpler that way

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**